drivers/interrupt_controller/Kconfig.shared_irq

```diff
@@ -5,6 +5,7 @@
 
 menuconfig SHARED_IRQ
 	bool "Shared interrupt driver"
+	default $(dt_compat_enabled,shared-irq)
```

Comment:
Since shared-irq can't be used alone, what about having this symbol selected by clients, when needed.